packages/apollo-gateway/src/buildQueryPlan.ts

```diff
@@ -827,7 +855,7 @@ export class QueryPlanningContext {
 
     const providedFields: FieldSet = [];
 
-    providedFields.push(...this.getKeyFields(returnType, serviceName));
+    providedFields.push(...this.getKeyFields(returnType, serviceName, true));
```

Comment:
I think this is going to need to filter more by the serviceName? I think it will end up overfetching otherwise

Reply:
The way I understand it, only the defining service can supply multiple independent keys, so only it will be affected by this change. And as the `providedFields` array is a list of all the fields that a service *can* provide without accessing any other service, it should include all the possible keys that the service knows.